src/System.Diagnostics.DiagnosticSource/src/DiagnosticSourceUsersGuide.md

```diff
@@ -0,0 +1,377 @@
+the DiagnosticListener implements.   DiagnositicListeners are given names when they are created
+and this name should be the name of logical grouping of related events (typically the component).
+Later this name is used to find the Listener and subscribe to any of its events.  
+
+Once you have an instance of a DiagnositicSource, logging is very straightforward.  The
+interface consists of only two methods.  
+
+```C#
+	bool IsEnabled(string name)
+	void Write(string name, object value);
+```
+
+A typical call site will look like
+
+```C#
+	if (httpLogger.IsEnabled("RequestStart"))
+		httpLogger.Write("RequestStart", new { Url="http://clr", Request=aRequest });
+```
+
+Already some of the architectural elements are being exposed namely
+
+	1. Every event has a string name (e.g. Request Start), and exactly one object as a payload.
+	2. If you need to send more than one item, you can do so by creating a object with all information
+	   in it as properties.   CSharp's [anonymous type](https://msdn.microsoft.com/en-us/library/bb397696.aspx)
+	   feature is typically used to create a type to pass 'on the fly', and makes this scheme very
+	   convenient.     
+	3. At the instrumentation site, you must guard the call to 'Write' with an 'IsEnabled' check on 
+       the same event name.   Without this check, even when the instrumentation is inactive, the rules
+	   of the C# language require all the work of creating the payload object and calling 'Write' to be 
+	   done, even though nothing is actually listening for the data. By guarding the 'Write' call, we 
+	   make it efficient when the source is not enabled.   
+
+### Creating DiagnosticSources (Actually DiagnosticListeners)
+	   
+The code above assumes there is an DiagnosticSource called 'mySource'.   Perhaps confusingly you 
+make a DiagnosticSource by creating DiagnosticListener
+
+	static DiagnosticSource mySource = new DiagnosticListener("System.Net.Http");
+
+Basically a DiagnosticListener is a named place where a source sends its information (events).  
+From an implementation point of view, DiagnosticSource is a abstract class that has the two
+instrumentation methods, and DiagnosticListener is something that implements that abstract class.
+Thus every DiagnosticListener is a DiagnosticSource.   
+
+DiagnosticListener have a name, which is used to represent the component associated with the event.
+Thus the event names only need to be unique within a component.  
+
+----------------------------------------
+## Best Practices
+
+### Naming Conventions
+
+#### DiagnositicListener Names
+
+ * CONSIDER - the likely scenarios for USING information when deciding how may 
+   DiagnositicListener to have and the events in each.   Keep in mind that it is **very easy
+   and efficient** to filter all the events in a particular listener so ideally the 
+   most important scnearios involve turning in whole listners.    You may need to split
```

Comment:
"involve turning in whole listeners"... I'm not sure what that means.

Nit: listners => listeners